streams/shared/src/main/scala/zio/stream/ZStream.scala

```diff
@@ -3148,6 +3142,14 @@ object ZStream extends ZStreamPlatformSpecificConstructors with Serializable {
   final def fromQueueWithShutdown[R, E, A](queue: ZQueue[Nothing, Any, R, E, Nothing, A]): ZStream[R, E, A] =
     fromQueue(queue).ensuringFirst(queue.shutdown)
 
+  /**
+   * Creates a stream from a [[Pull]] producing a value of type `A` or an empty Stream
+   */
+  final def fromPull[R, E, A](pull: Pull[R, E, A]): ZStream[R, E, A] =
```

Comment:
Possible alternative name: `fromEffectOption`. More descriptive and won't result in silent breakage for anyone upgrading.

e.g.
```suggestion
  final def fromEffectOption[R, E, A](effect: ZIO[R, Option[E], A]): ZStream[R, E, A] =
```

wdyt? @adamgfraser @vasilmkd

Reply:
Agreed, we must assume this is already in use.

Reply:
It would be nice to have consistency between the repeating and single versions of this and `effect` but your point about breakage is a strong one. One other thought I had is whether users have an understanding of what a `Pull` is. The type alias is extremely useful for us internally but if I wanted to use an `Option` to express stopping would I know what the `Pull` type alias is and that is what I should use? Maybe there is a more descriptive name for both combinators? Just an idea. I’m good with any of these.

Reply:
Yes, I agree that we should probably not expose `Pull`.

`fromEffectOption` and `repeatEffectOption` check those marks for me.

Reply:
Agreed.